app/bundles/LeadBundle/Assets/js/lead.js

```diff
@@ -389,7 +390,7 @@ Mautic.reorderSegmentFilters = function() {
             }
 
             var newName = prefix+'[filters]['+counter+']['+suffix+']';
-            if (name.slice(-2) === '[]') {
+            if (name !== undefined && name.slice(-2) === '[]') {
```

Comment:
```suggestion
            if (typeof name !== 'undefined' && name.slice(-2) === '[]') {
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined with the section called "DON'T DO THIS"

Reply:
"DON'T DO THIS" is about scope of undefined.

`name !== undefined` and `typeof name !== 'undefined'` is exactly same (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#Strict_equality_and_undefined)

The one reason to use typeof is that it does not throw an error if the variable has not been declared, and it's declared L. 385

Reply:
The problem is that `undefined` is not a reserved keyword. So anyone can define a variable called `undefined` in a parent scope and break your code.